### PR TITLE
Teleport kube agent

### PR DESCRIPTION
**Pull request by ssyno**
### What this PR does / why we need it

- This PR installs `teleport-kube-agent` by default on EKS clusters.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: helm/default-apps-eks/values.yaml [hunk 1]
```diff
@@ -131,6 +131,19 @@ apps:
     # used by renovate
     # repo: giantswarm/cluster-autoscaler-app
     version: 1.27.3-gs5
+  coreDNS:
```
Comment (AndiDog) [inline helm/default-apps-eks/values.yaml:134]: We already install [coredns in the cluster chart](https://github.com/giantswarm/cluster-eks/blob/main/helm/cluster-eks/templates/coredns-helmrelease.yaml)

**Review by AndiDog — CHANGES_REQUESTED** [review]

File: helm/default-apps-eks/values.yaml (hunk 1 above)
Comment (ssyno) [reply helm/default-apps-eks/values.yaml:134]: oh nice, I will remove it, what about its configuration?

**Review by tuladhar — APPROVED** [review]

File: helm/default-apps-eks/templates/apps.yaml [hunk 2]
```diff
@@ -49,8 +49,16 @@ spec:
       name: {{ $.Values.clusterName }}-cluster-values
       namespace: {{ $.Release.Namespace }}
   {{- end }}
-  {{- end }} 
+  {{- end }}
 {{- end }}
+  {{- if .extraConfigs }}
```
Comment (AndiDog) [inline helm/default-apps-eks/templates/apps.yaml:54]: Let's make this consistent with default-apps-aws where we support the `priority` feature as of a customer request.

See https://github.com/giantswarm/default-apps-aws/blob/d7a1dd710447356920767d7f7d9a44b55be1af38/helm/default-apps-aws/templates/apps.yaml#L54-L63 (but use correctly-spelled `priority`, not `priotity` – that's to be fixed over there).

File: helm/default-apps-eks/values.yaml [hunk 3]
```diff
@@ -207,7 +207,26 @@ apps:
...
+    namespace: kube-system
+    # used by renovate
+    # repo: giantswarm/teleport-kube-agent-app
+    version: 0.7.1
+    # a list of extraConfigs for the App,
+    # It can be secret or configmap
+    # https://github.com/giantswarm/rfc/tree/main/multi-layer-app-config#example
+    extraConfigs:
+      - kind: configMap
+        name: "{{ $.Values.clusterName }}-teleport-kube-agent-config"
```
Comment (AndiDog) [inline helm/default-apps-eks/values.yaml:228]: Please document here where the config comes from (seems to be externally introduced, I guess by an operator).

File: helm/default-apps-eks/templates/apps.yaml (hunk 2 above)
Comment (ssyno) [reply helm/default-apps-eks/templates/apps.yaml:54]: sure, updating itnow

File: helm/default-apps-eks/values.yaml (hunk 3 above)
Comment (ssyno) [reply helm/default-apps-eks/values.yaml:229]: Added comment

**Review by calvix — APPROVED** [review]

**Review by AndiDog — COMMENTED** [review]
Please go ahead and merge once the `priotity` typo is fixed

**Review by AndiDog — APPROVED** [review]

**Pull request merged**